sentry-android-core/src/main/java/io/sentry/android/core/UserInteractionIntegration.java

```diff
@@ -0,0 +1,183 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.os.Build;
+import android.os.Bundle;
+import android.view.Window;
+import io.sentry.IHub;
+import io.sentry.Integration;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.android.core.gestures.NoOpWindowCallback;
+import io.sentry.android.core.gestures.SentryGestureListener;
+import io.sentry.android.core.gestures.SentryWindowCallback;
+import io.sentry.util.Objects;
+import java.io.Closeable;
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class UserInteractionIntegration
+  implements Integration, Closeable, Application.ActivityLifecycleCallbacks {
+
+  private final @NotNull Application application;
+  private @Nullable IHub hub;
+  private @Nullable SentryAndroidOptions options;
+
+  private final boolean isAllActivityCallbacksAvailable;
+  private final boolean isAndroidXAvailable;
+
+  public UserInteractionIntegration(
+    final @NotNull Application application,
+    final @NotNull IBuildInfoProvider buildInfoProvider,
+    final @NotNull LoadClass classLoader
+  ) {
+    this.application = Objects.requireNonNull(application, "Application is required");
+    Objects.requireNonNull(buildInfoProvider, "BuildInfoProvider is required");
+
+    isAndroidXAvailable = checkAndroidXAvailability(classLoader);
+    isAllActivityCallbacksAvailable =
+      buildInfoProvider.getSdkInfoVersion() >= Build.VERSION_CODES.Q;
+  }
+
+  private static boolean checkAndroidXAvailability(final @NotNull LoadClass loadClass) {
+    try {
+      loadClass.loadClass("androidx.core.view.GestureDetectorCompat");
+      return true;
+    } catch (ClassNotFoundException ignored) {
+      return false;
+    }
+  }
```

Comment:
Is there a way to know if the `androidx.core` package exists instead of a specific class? so we could do this check only once and pass the result along with the integrations.

Reply:
not that I know of. We could just check for some random class from androidx.core, that we know should be there and assume that the entire package exists. But what if there's a version, where one of those classes didn't exist? we'd crash in this case

Reply:
yeah indeed, guess its better like this for now